pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -394,6 +394,12 @@ def _detect_replacable_format_call(self, node: nodes.Const) -> None:
             if "\\" in node.parent.right.as_string():
                 return
 
+            # If % applied to another type than str, it's modulo and can't be replaced by formatting
+            if not hasattr(node.parent.left, "value") or not isinstance(
```

Comment:
Should we infer the left-hand side (e.g. like the right is inferred just below) to avoid creating the false negatives reported by the primer?

Reply:
Is this really false negative ? It's really hard to know if a string is supposed to be a template or not. 

```python
templates = {
    "config": "blabla %s blabla"
}

if some_condition:
    print(templates["config"} % "a string")
else:
    print(templates["config"} % "another string")
```
Is often seen and I don't think we should raise here. It's the reason why we do not have a check for missing leading f in supposed f-string (too much false positive).

Reply:
Good point!